text/0314-cdk-construct-hub.md

```diff
@@ -362,6 +367,96 @@ pipeline.
 The details of URLs to be configrued in CloudFront is available in [appendix
 "CloudFront URLs"](#cloudfront-urls).
 
+#### Backend X Frontend Compatibility
+
+As described earlier, the front-end application consumes artifacts that are stored on the backend. Since these artifacts are subject to change,
+we need to define a mechanism by which the front-end is able to detect and operate on various versions of them.
+
+To that end, we propose a generic mechanism to apply to all artifact types:
+
+Every artifact will comply to the [SemVer](https://semver.org/spec/v2.0.0.html) specification
+and include its major version in the `version` field of the schema:
+
+```json
+{
+  "version": "v1",
+  ...
+}
+```
+
+```json
+{
+  "version": "v2",
+  ...
+}
+```
+
+Every time a change in the artifact occurs, the contents of the artifact is replaced with the new version.
+This means that at any given point in time, the artifact can either be in its new version, or the old one.
```

Comment:
Thanks, yes I indeed meant *an older one*. Changed to:

*Since artifacts are created in a long running background process, the new version may not be available immediately. This in turn means that at any given point in time, the artifact can either be in its new version, or an older one.*